### ISSUE TITLE
Provide option in installer to not mess with default shell environment

### ISSUE THREAD

**Issue by andrej** [conda/conda#8677]
This concerns the installer script for Miniconda on macOS (and possibly other platforms, not tested).

**Current behavior:**

After installation of Miniconda with the installer script on macOS, the binaries for python, pip, etc, are shadowed in the PATH by the Conda ones. Also, the string "(base)" is prepended in front of the prompt.

Of course, this can be reversed by running `conda deactivate`. However, frankly, I think it is a little arrogant that the Conda environment is *activated by default for every single shell* opened. It might also break some things, e.g. for programs and scripts relying on `python` pointing to macOS's default python interpreter, not Conda's.

**Suggested new behavior:**

 1. By default, **do not touch** `~/.bash_profile`. Put that stuff somewhere else, e.g. `~/miniconda3/.bash_profile`.
 2. Instead, to enable the conda environment, let users run `bash --rcfile ~/miniconda3/.bash_profile`. 
(Maybe, for convenience, implement a script named `conda-shell` or something that does exactly this to enter the Conda environment.)

I would expect this as the default behavior. If that's not possible, at least having an option to not alter `~/.bash_profile` would be nice. At the very least warn users about the fact that several important binaries will be shadowed after installation in their shell!

**To summarize, ...**

The only reason I installed Conda is because it is required by *one single project* I am working on. For all the others, I want to keep things as they are. Coming from that perspective, you might understand why I found it rather irritating that, after installation, all those binaries were shadowed in *every shell* and I had to look at that annoying new prompt string.

**Comment by msarahan** [comment]
You are given the option to not do this modification during install.  Did you not see that?

**Comment by andrej** [comment]
I did not, but do now. I apologize! This can be closed.
